File: src/veiculos/dto/create-veiculo.dto.ts
import { ApiProperty } from '@nestjs/swagger';
export class CreateVeiculoDto {
  @ApiProperty({ type: String, description: 'Placa como identificador do veiculo.' })
  readonly placa: string;
  @ApiProperty({ type: String, description: 'Tipo do veículo [automovel] ou [motocicleta] ' })
  readonly tipo: string;
  @ApiProperty({ type: String, description: 'Marca do veículo' })
  readonly marca: string;
  @ApiProperty({ type: String, description: 'Modelo do veiculo' })
  readonly modelo: string;
  @ApiProperty({ type: String, description: 'Cor do veiculo' })
  readonly cor: string;
}

File: src/main/java/com/fcamarasantos/testebackendjava/domain/veiculo/dto/CreateVeiculoDTO.java
package com.fcamarasantos.testebackendjava.domain.veiculo.dto;

import com.fcamarasantos.testebackendjava.domain.veiculo.TipoVeiculo;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

public record CreateVeiculoDTO(
        @NotBlank
        String marca,

        @NotBlank
        String modelo,

        @NotBlank
        String placa,

        @NotBlank
        String cor,

        @NotNull
        TipoVeiculo tipo
) {
}
